fab/fabfile.py

```diff
@@ -107,66 +109,49 @@ def _require_target():
 
 
 class DeployMetadata(object):
-    def __init__(self, env):
-        self.env = env
+    def __init__(self):
         self.timestamp = datetime.datetime.utcnow().strftime('%Y-%m-%d_%H.%M')
         self._deploy_ref = None
         self._deploy_tag = None
-        self._git = sh.git.bake(_tty_out=False)
+        self._repo = GitHub().repository('dimagi', 'commcare-hq')
+        self._max_tags = 100
+        self._last_tag = None
 
     def tag_commit(self):
-        self._git.fetch("origin", "--tags")
-        pattern = "*{}*".format(self.env.environment)
-        self.last_tag = sh.tail(self._git.tag("-l", pattern), "-1").strip()
-
-        tag_name = "{}-{}-deploy".format(self.timestamp, self.env.environment)
-        msg = "{} deploy at {}".format(self.env.environment, self.timestamp)
-        self._git.tag(tag_name, "-m", msg, self.deploy_ref)
-        self._git.push("origin", tag_name)
+        pattern = ".*-{}-.*".format(env.environment)
+        for tag in self._repo.tags(self._max_tags):
+            if re.match(pattern, tag.name):
```

Comment:
yeah the docs said `iter_tags` but when i was doing it locally it said `iter_tags` wasn't a function and `tags` was haha. no i don't think the api has the pattern arg that i'm aware of: https://developer.github.com/v3/git/refs/#get-all-references. but the api is huge so maybe it can be done with a different call